### ISSUE TITLE
`deploy` messages wrong for remotes that aren't ssh

### ISSUE THREAD

**Issue by oscarlevin** [PreTeXtBook/pretext-cli#350]
@mitchkeller points out that if you have cloned a github repo using https instead of ssh, then the messages you get when you deploy do not include the username or repo name (since those are being detected based on the file pattern for ssh).

**Comment by StevenClontz** [comment]
Relatedly, I understand that HTTPS is supported with secret keys. I think that's a much cleaner workflow for non-dev users that avoids SSH setup cruft, so we really should make that the default path (while continuing to support SSH keys).

See https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

and maybe https://github.com/GitCredentialManager/git-credential-manager/blob/main/README.md

**Comment by mitchkeller** [comment]
Just chiming in here to say that I'm glad that @oscarlevin was wrong when he said that @StevenClontz would say that everyone should be using SSH keys 😉

I have been using personal access tokens for a while now, but with my new work laptop, I set up the git credential manager, and it was much simpler than the personal access token route. (Since I had homebrew installed, I think it was literally a brew install command and then maybe one other command and it was done.) I think it's meant to be cross-platform, so I would probably encourage folks to go that route.

**Comment by StevenClontz** [comment]
Advantage of PAA is not having another dependency. And I'm not sure about
git credential manager when a GUI web browser isn't available, but it
should be investigated.

**Comment by oscarlevin** [comment]
I guess we are now talking about two issues: first, what should we recommend in the documentation and in the automated setup when `pretext deploy` is first run, and second, how should we determine what to output when `deploy` is run.  I would be in favor of making sure that no matter how github is set up, we log the correct repository names.

**Comment by StevenClontz** [comment]
I guess one question is: if one uses git cred manager, what does the `origin` remote look like for the repo? Is it an SSH, HTTPS, or some other string?

**Comment by mitchkeller** [comment]
`git remote -v` with git credential manager reports:

origin	https://github.com/mitchkeller/linear-algebra.git (fetch)
origin	https://github.com/mitchkeller/linear-algebra.git (push)

Since the CLI is giving the same output on my personal laptop that’s using a personal access token, I think that origin is set up the same.

**Comment by oscarlevin** [comment]
Right.  It looks like the two patterns are

```
https://github.com/user/repo.git
git@github.com:user/repo.git
```

according to https://docs.github.com/en/get-started/getting-started-with-git/about-remote-repositories

**Comment by StevenClontz** [comment]
So to summarize what I think we all now know: if we support parsing SSH and HTTPS strings, we're good to infer info about the repo. The only thing left is to decide on the steps we walk users through when they first try to `pretext deploy`.

**Comment by oscarlevin** [comment]
The original issue here is now fixed (at #351).  The instructions now ask the user to paste the github repo url, in either https or ssh, and we give a link to the github authentication help page.

Eventually we can consider offering to manage the user's github credentials, but that is a bigger discussion, and we should see about feedback from our current setup to see if it is needed.  I'll make a new issue for that so we don't lose the idea.
